### PR TITLE
fix(connection-manager): mssql datatype parsing

### LINKED ISSUE #9468: Sequelize returns timestamps on DATEONLY fields
## What are you doing?
I am using mssql.

My DATEONLY fields are being stored as YYYY-MM-DD, but when I pull that data down to the client, it is being sent as a full timestamp with time and timezone parts.

I can try to workaround this with a custom get() method for each DATEONLY field, but things like changed() will still detect changes.

## What do you expect to happen?
Data from DATEONLY fields to be returned without time (e.g. "2016-02-13")

## What is actually happening?
Data from DATEONLY fields is being returned as with timestamp (e.g. "2016-02-13T00:00:00.000Z")

__Dialect:__ mysql
__Dialect version:__ tedious 2.6.2
__Database version:__ MS SQL Server 2017 (14.0.1000.169)
__Sequelize version:__ 4.37.8
__Tested with latest release:__ Yes (If yes, specify that version) 4.37.8

**Note :** _Your issue may be ignored OR closed by maintainers if it's not tested against latest version OR does not follow issue template._

### PR DESCRIPTION
**Pull request by sushantdhiman**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](../CONTRIBUTING.md)?

### Description of change

Fixes #9468

Not sure now tests did not caught this, will check CI with this PR

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -30,6 +27,7 @@ before_test:
   - ps: . .\appveyor-setup.ps1
 
 test_script:
+  - npm ls tedious
```
Comment (eseliger) [inline appveyor.yml]: is this related?

File: appveyor.yml [hunk 2]
```diff
@@ -8,12 +8,9 @@ services:
 
 shallow_clone: true
 
-cache:
```
Comment (eseliger) [inline appveyor.yml]: this was causing the problem? 🤔

Comment (sushantdhiman) [reply appveyor.yml]: Appveyor was loading `tedious@2.3.1` for which I think everything was working properly as same tests will fail locally with `2.6.1`.

So I have removed caching entirely, as this is single test suite it should easily finish before its travis counterpart.

File: appveyor.yml (hunk 1 above)
Comment (sushantdhiman) [reply appveyor.yml]: Debugging, will remove
